domain/status/state/crossmodelrelation.go

```diff
@@ -0,0 +1,207 @@
+	ctx context.Context,
+	name string,
+) (coreremoteapplication.UUID, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	var uuid coreremoteapplication.UUID
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var err error
+		uuid, err = st.lookupRemoteApplicationOfferer(ctx, tx, name)
+		return err
+	}); err != nil {
+		return "", errors.Capture(err)
+	}
+
+	return uuid, nil
+}
+
+func (st *ModelState) GetRemoteApplicationOffererStatus(
+	ctx context.Context,
+	uuid string,
+) (status.StatusInfo[status.WorkloadStatusType], error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return status.StatusInfo[status.WorkloadStatusType]{}, errors.Capture(err)
+	}
+
+	ident := remoteApplicationUUID{RemoteApplicationUUID: uuid}
+
+	query, err := st.Prepare(`
+SELECT &statusInfo.*
+FROM application_remote_offerer_status
+WHERE application_remote_offerer_uuid = $remoteApplicationUUID.uuid
+`, statusInfo{}, ident)
+	if err != nil {
+		return status.StatusInfo[status.WorkloadStatusType]{}, errors.Capture(err)
+	}
+
+	var statusInfo statusInfo
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err := st.checkRemoteApplicationNotDead(ctx, tx, ident); err != nil {
+			return errors.Capture(err)
+		}
+		err := tx.Query(ctx, query, ident).Get(&statusInfo)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			// If the application status is not set, we should return a default unset status
+			// instead of an error.
+			return nil
+		} else if err != nil {
+			return errors.Capture(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return status.StatusInfo[status.WorkloadStatusType]{}, errors.Capture(err)
+	}
+
+	statusType, err := status.DecodeWorkloadStatus(statusInfo.StatusID)
+	if err != nil {
+		return status.StatusInfo[status.WorkloadStatusType]{}, errors.Capture(err)
+	}
+
+	return status.StatusInfo[status.WorkloadStatusType]{
+		Status:  statusType,
+		Message: statusInfo.Message,
+		Data:    statusInfo.Data,
+		Since:   statusInfo.UpdatedAt,
+	}, nil
+}
+
+func (st *ModelState) SetRemoteApplicationOffererStatus(
+	ctx context.Context,
+	remoteApplicationUUID string,
+	sts status.StatusInfo[status.WorkloadStatusType],
+) error {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	statusID, err := status.EncodeWorkloadStatus(sts.Status)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	statusInfo := remoteApplicationStatus{
+		RemoteApplicationUUID: remoteApplicationUUID,
+		StatusID:              statusID,
+		Message:               sts.Message,
+		Data:                  sts.Data,
+		UpdatedAt:             sts.Since,
+	}
+
+	stmt, err := st.Prepare(`
+INSERT INTO application_remote_offerer_status (*) VALUES ($remoteApplicationStatus.*)
+ON CONFLICT(application_remote_offerer_uuid) DO UPDATE SET
+    status_id = excluded.status_id,
+    message = excluded.message,
+    updated_at = excluded.updated_at,
+    data = excluded.data;
```

Comment:
So there will never be data for remote application, I don't think it matters in the end.